### PR TITLE
Vidur/tf

### PR DESCRIPTION
**Pull request by vidurvij-Unity**
## Proposed change(s)

Adding code to insatiate Joint_state sensors and transform sensors in a robot during import.

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

Provide any relevant links here.

### Types of change(s)

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe)

## Testing and Verification

Please describe the tests that you ran to verify your changes. Please also provide instructions, ROS packages, and Unity project files as appropriate so we can reproduce the test environment. 

### Test Configuration:
- Unity Version: [e.g. Unity 2020.2.0f1]
- Unity machine OS + version: [e.g. Windows 10]
- ROS machine OS + version: [e.g. Ubuntu 18.04, ROS Noetic]
- ROS–Unity communication: [e.g. Docker]

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [ ] Created this PR to target the `dev` branch
- [ ] Followed the style guidelines as described in the [Contribution Guidelines](https://github.com/Unity-Technologies/URDF-Importer/blob/main/CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Increased the [test coverage criteria](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/.yamato/yamato-config.yml#L18) by 3%
- [ ] Updated the [Changelog](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/com.unity.robotics.urdf-importer/CHANGELOG.md) and described changes in the [Unreleased section](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/com.unity.robotics.urdf-importer/CHANGELOG.md#unreleased)
- [ ] Updated the documentation as appropriate

## Other comments

### CODE REVIEW

**Review by hyounesy — COMMENTED** [review]
Why is this a PR to vidur/SensorImport rather than dev-sensors?

File: com.unity.robotics.urdf-importer/Runtime/UnityMeshImporter/Plugins/AssimpNet/AssimpNet.xml [hunk 1]
```diff
@@ -2209,7 +2209,7 @@
         </member>
         <member name="T:Assimp.Configs.IRRAnimationFrameRateConfig">
             <summary>
-            Defines the output frame rate of the IRR loader.
+            Defines the Output frame rate of the IRR loader.
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/UnityMeshImporter/Plugins/AssimpNet/AssimpNet.xml:2212]: revert this file. This looks like an unintended consequence of doing a mass find-replace.

File: com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/Joint.cs [hunk 2]
```diff
@@ -150,7 +150,7 @@ public void WriteToUrdf(XmlWriter writer)
             {
                 writer.WriteStartElement("calibration");
 
-                //TODO only output one or the other
+                //TODO only Output one or the other
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/Joint.cs:153]: revert this file. This looks like an unintended consequence of doing a mass find-replace.

File: com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfRobotExtensions.cs [hunk 3]
```diff
@@ -385,6 +387,10 @@ public static void CreateTag()
 #endif
         }
 
+        static void AddJointSensor(GameObject robot)
+        {
+            Dictionary<string, string> settings = new Dictionary<string, string> { { "sensor/topic", robot.name + "/JointState"} };
+            SensorFactory.InstantiateSensor("joint", settings).transform.SetParentAndAlign(robot.transform);
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfRobotExtensions.cs:393]: I am not sure if this compiles. Bad Merge?

**Review by hyounesy — CHANGES_REQUESTED** [review]
- [ ] Make this a PR to dev-sensors.
- [ ] Revert un-wanted changes 
- [ ] potential merge issue in UrdfSensorExtensions.cs

File: com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfRobotExtensions.cs [hunk 4]
```diff
@@ -322,14 +335,16 @@ private static Robot ExportRobotData(this UrdfRobot urdfRobot)
...
                 robot.links.Add(urdfLink.ExportLinkData());
                 linkNames.Add(urdfLink.name);
 
-                //Joint export
+                //Joints export
                 UrdfJoint urdfJoint = urdfLink.gameObject.GetComponent<UrdfJoint>();
                 if (urdfJoint != null)
                     robot.joints.Add(urdfJoint.ExportJointData());
                 else if (!urdfLink.IsBaseLink)
+
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfRobotExtensions.cs:347]: remove black line. might also be good to enclose the else body in brackets as it is more than 1 line even though one is a comment.

File: com.unity.robotics.urdf-importer/Runtime/Unity.Robotics.URDFImporter.asmdef [hunk 5]
```diff
@@ -2,7 +2,8 @@
     "name": "Unity.Robotics.UrdfImporter",
     "rootNamespace": "Unity.Robotics.UrdfImporter.Urdf",
     "references": [
-        "GUID:6a697808d7c80a549b57420070d6c4f3"
+        "GUID:6a697808d7c80a549b57420070d6c4f3",
+        "GUID:7c3cd0727030049da8a847e9ce61c46c"
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/Unity.Robotics.URDFImporter.asmdef:6]: uncheck the "use GUID"  in the asmdef inspector. to make it clear the dependencies.

File: com.unity.robotics.urdf-importer/Runtime/UnityMeshImporter/Plugins/AssimpNet/AssimpNet.xml [hunk 6]
```diff
@@ -3735,7 +3735,7 @@
         <member name="F:Assimp.DefaultLogStream.Debugger">
             <summary>
             MSVC only: Stream the log to the debugger (this relies
-            on OutputDebugString from the Win32 SDK).
+            on outputDebugString from the Win32 SDK).
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/UnityMeshImporter/Plugins/AssimpNet/AssimpNet.xml:3738]: revert all modifications made to AssimpNet.xml. (grab the original unchanged one from "dev" branch?)

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfComparator.cs [hunk 7]
```diff
@@ -167,7 +167,7 @@ private bool CompareLink( Link source,  Link exported, int indent)
                 Joint jointExported = exported.joints.Find(x => x.name == jointSource.name); // Check for no match
                 if (jointExported == null)
                 {
-                    linkLog.AppendLine(String.Format("{0}Joint Not Found in Exported: Joint Name:{1,12}",Indent(indent + 1),jointSource.name));
+                    linkLog.AppendLine(String.Format("{0}Joints Not Found in Exported: Joints Name:{1,12}",Indent(indent + 1),jointSource.name));
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfComparator.cs:170]: nit: space before Joints: "{0} Joints"
coding style: space after comma

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensors.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+using UnityEngine;
+
+namespace Unity.Robotics.UrdfImporter
+{
+    public class UrdfSensors:MonoBehaviour
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensors.cs:5]: coding style: space after colon :
Also, Is this class used anywhere? what is the purpose of it?

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensor.cs [hunk 9]
```diff
@@ -0,0 +1,9 @@
+using UnityEngine;
+
+namespace Unity.Robotics.UrdfImporter
+{
+    public class UrdfSensor: MonoBehaviour
```
Comment (hyounesy) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensor.cs:5]: Is this class used anywhere?

**Review by hyounesy — CHANGES_REQUESTED** [review]

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensors.cs (hunk 8 above)
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensors.cs:5]: https://github.com/Unity-Technologies/URDF-Importer/blob/vidur/tf/com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfSensorsExtensions.cs

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensor.cs (hunk 9 above)
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfSensor.cs:5]: https://github.com/Unity-Technologies/URDF-Importer/blob/vidur/tf/com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfSensorExtension.cs

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfComparator.cs (hunk 7 above)
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfComparator.cs]: Its an indent, doesn't need space.

**Review by hyounesy — APPROVED** [review]

**Pull request merged**